danascheider/skyrim_inventory_management_frontend
# Add games page to the dashboard

## Context

[**Add Games page to the dashboard**](https://trello.com/c/uioZOfvp/90-add-games-page-to-the-dashboard)

On the backend, we've implemented a [feature](https://github.com/danascheider/skyrim_inventory_management/pull/32) whereby a user can have multiple games. Now, instead of belonging to a user directly, all shopping lists are scoped to the game they were created for. This means that the shopping list page is now broken while we update the front end to account for this. It also means we need a page on the dashboard where the user can view all their games. This functionality is both more straightforward than fixing the shopping lists and also necessary to test changed shopping list functionality in production, since users will now need to create a game before they can create shopping lists.

Game creation is not part of this PR, however, in order to create a game, there first needs to be a page where that functionality will live. This PR adds that page and, in the meantime, removes the shopping list page entirely.

## Changes

* Add `GamesPage` component
* Define `/dashboard/games` route in page routes file
* Add `Game` component as list item and incorporate into games page
* Remove route for shopping list page (all components, including the page itself, still exist and are functional in Storybook)
* Add Storybook stories for new components
* Add Jest tests for the `GamesPage` component

## Considerations

Since the page appearance will change somewhat as additional elements are added, I've decided not to worry too much about making the page look impressive for this card. I did add slide toggling for the game descriptions and make a decent looking list view, but I expect it will change a bit as we get further along with this feature.

## Manual Test Cases

At this point, the only way to view the page (and the `Game` component) in all its states is in Storybook, unless you want to manually create games in the dev environment, which I didn't feel like doing. However, you can visit the dashboard page and make sure that (1) the link to the Games page takes you to the Games page (you should see a loading component and then a "You have no games" message at the top) and (2) the link to the Shopping List page doesn't do anything.

## Screenshots and GIFs

Most of these only show the body of the page and leave the header out, because they were taken in Storybook and the header is a little messed up in there.

### View with No Games

<img width="1678" alt="Screen Shot 2021-07-18 at 2 56 11 pm" src="https://user-images.githubusercontent.com/5115928/126056384-08407cc5-50d4-4107-9110-c302c5360bc5.png">

### View with Games (Wide Screen)

<img width="1368" alt="Screen Shot 2021-07-18 at 2 59 36 pm" src="https://user-images.githubusercontent.com/5115928/126056393-4215e0bb-24b7-463c-b970-e0b56bf8c5c8.png">

### View with Games (Medium Screen)

<img width="930" alt="Screen Shot 2021-07-18 at 2 59 49 pm" src="https://user-images.githubusercontent.com/5115928/126056405-02487184-eb30-4b92-b19d-93cde38dc379.png">

### View with Games (Small Screen)

<img width="498" alt="Screen Shot 2021-07-18 at 3 00 03 pm" src="https://user-images.githubusercontent.com/5115928/126056407-4b90338d-51b0-49ab-b2be-d6675c7bbf50.png">

### View with Description Expanded (Small Screen)

<img width="495" alt="Screen Shot 2021-07-18 at 3 00 29 pm" src="https://user-images.githubusercontent.com/5115928/126056414-478a7384-e274-4b88-8740-6b5126763c95.png">
